### PR TITLE
Endpoint testing

### LINKED ISSUE #4: Test Endpoints

### PR DESCRIPTION
**Pull request by GreyMatteOr**
# Description

Create testing environment for API-server. Also tested some happy and sad for each of:
  - `joinGame`
  - `setWordToGuess`
  - `makeGuess`

Also implemented `cors` so hosting on `heroku` could modify data from the outside.
# To Run tests

Navigate to root directory in terminal. Use command `npm run jest server.test.js`

# Modifications to dependencies

brought in `supertest` using `npm i --save-dev supertest`
   - this is for making requests inside the test

brought in `cors` using `npm i -save cors`
  - this is for hosting on heroku

# Snapshot

Screen Shot 2020-10-28 at 4 57 47 PM

# Linked Issues
(Issue #27)[https://github.com/GreyMatteOr/bangwords/issues/27]

Closes #4

### CODE REVIEW

**Review by holladayian — COMMENTED** [review]
I don't know if there were any sad paths that we're being tested. I know I'm also not great at this. Maybe we could all go back together and brainstorm some sad paths to test.

File: package.json
Comment (holladayian) [inline package.json:43]: woo

File: server.test.js
Comment (holladayian) [inline server.test.js:91]: The aforementioned sad path!!

File: server.test.js
Comment (holladayian) [inline server.test.js:132]: I've always wanted to put sad path testing into it's own describe block for visibility sake. Or maybe we could organize it some other way. What do you think? Got any ideas?

File: server.test.js
Comment (holladayian) [inline server.test.js:149]: Were we also going to add something like `attempts` into the response?

**Review by holladayian — APPROVED** [review]
Looks good to me! Just a couple clarifying questions. Should be good to merge though!

**Comment by holladayian** [comment]
Looks good to me! I also added a few labels, and created and closed and issue into this PR. It's got my approval. I'll let @Chriscastanuela take a look and merge.

File: server.test.js
Comment (GreyMatteOr) [reply server.test.js]: I would say any part of the res that we're relying on to have functioning display.  At the moment, I cannot think of anything. 

That being said, this brings up a great point that we should be paying attention to see if anything like that shows up.

File: server.test.js
Comment (GreyMatteOr) [reply server.test.js:32]: This is some experimental code I would like to have handy for when we get into testing `sockets`.

File: server.test.js
Comment (GreyMatteOr) [reply server.test.js:132]: Hmmm, interesting, I've never thought to do it that way. 

As I'm writing it at least, I feel when I'm writing tests for a single section, it feels easier to write the tests all at once. That way, I don't have to re-familiarize with the content and the ins and outs of what I'm dealing with. I feel like (in this case any way) that it's easier to procedurally go through the pieces and break them one by one to generate sad-paths. 

I'm very curious what doing it the other way might feel like!

File: server.test.js
Comment (GreyMatteOr) [reply server.test.js:149]: We sure could! Let me know if that's something that feels good to do.

**Pull request merged**
